### PR TITLE
Deployment contracts in grace period

### PR DESCRIPTION
**Pull request by samaradel**
### Description

Prevent deployment on nodes with contracts in the grace period.

### Changes
[Screencast from 31-07-24 16:38:10.webm](https://github.com/user-attachments/assets/983793f1-9267-4533-8759-d614029e23c6)

### Related Issues

- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/3025

### Checklist

- [ ] Tests included
- [x] Build pass
- [ ] Documentation
- [x] Code format and docstrings
- [x] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

File: packages/playground/src/components/node_selector/TfAutoNodeSelector.vue [hunk 1]
```diff
@@ -286,6 +286,14 @@ export default {
 
     const nodeInputValidateTask = useAsync<true, string, [NodeInfo | undefined]>(
       async node => {
+        if (node && node?.rentContractId !== 0) {
+          const { state } = await gridStore.grid.contracts.get({
+            id: node?.rentContractId,
+          });
+          if (state.gracePeriod) {
+            return false as true;
```
Comment (0oM4R) [inline packages/playground/src/components/node_selector/TfAutoNodeSelector.vue]: why we should do that? should we just modify the return type from true to boolean?

Comment (samaradel) [reply packages/playground/src/components/node_selector/TfAutoNodeSelector.vue]: No, as useAsync should return true.. you can try it.

Comment (0oM4R) [reply packages/playground/src/components/node_selector/TfAutoNodeSelector.vue]: image
compiled successfully, unable to test the functionality now,, will try again whenever the chain transfer got fixed

**Comment by Mahmoud-Emad** [comment]
I would recommend excluding these nodes from the filters
also, if this is important info that the user should know, we can display a dialog with the nodes that are excluded. 

But, I think it's not a good behavior to tell the user that this node isn't available after filtering, he user needs to deploy with only a single click.

**Comment by samaradel** [comment]
@Mahmoud-Emad I think we had an issue for displaying them before, @AhmedHanafy725 what do you think?

**Comment by AhmedHanafy725** [comment]
IMO, we should show the node to him and notify him that the contract is in a grace period instead of hiding it and he will keep looking for why it's not on the list

**Review by maayarosama — APPROVED** [review]

**Review by 0oM4R — APPROVED** [review]

**Pull request merged**